### PR TITLE
Add keywords in comments metric

### LINKED ISSUE #298: Metric number for occurence of certain keywords in comments
see https://github.com/MaibornWolff/codecharta/issues/281

### LINKED ISSUE #315: Refactor CommentLines metric
We need to refactor the `src/parser/metrics/CommentLines.ts`:
- Use `SimpleLanguageSpecificQueryStatement` instead of `SimpleQueryStatement` for block comment query (Python). `SimpleQueryStatement` does not specify the language to query on and is very error prone. We may remove it entirely.
- add function `addQueryForPython` and use it in the constructor instead of proving `parseFile.language` in `calculate(...)`
- use `#statementsSuperSet` instead of `private statementsSuperSet`, since the last one does not transpile the variable to private correctly. We may change that in every metric.
- use `query.capture` instead of `query.match`, since we only have a capture for each match (YAGNI)

### PR DESCRIPTION
**Pull request by ce-bo**
# Add keywords in comments metric
Closes #298 
Closes #315 

## Description
- Added new metric **keywords in comment**: return the number of theses keywords in comments with case insensivity: wtf, todo, hack, bug.
- Refactor the CommentLines metric to make it more inline with other metrics and to prevent errors. 
- Documented the new metric in `README.md`
- Add tests for the new metric
## Definition of Done

A PR is only ready for merge once all the following acceptance criteria are fulfilled:
* [x]  There are automated tests for newly written code and bug fixes
* [x]  Documentation ([README.md](https://github.com/MaibornWolff/metric-gardener/blob/main/README.md), [UPDATE_GRAMMARS.md](https://github.com/MaibornWolff/metric-gardener/blob/main/docs/UPDATE_GRAMMARS.md)) has been updated

### CODE REVIEW

File: src/parser/metrics/CommentKeyword.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    dlog = logger;
+});
+
+export class CommentKeyword implements Metric {
+    private readonly statementsSuperSet: QueryStatementInterface[] = [];
+
+    private readonly keywords: string[] = [
+        "BUG",
+        "Bug",
+        "bug",
```
Comment (ce-bo) [inline src/parser/metrics/CommentKeyword.ts:17]: Let's refactor this to check keywords case insensitive.

File: src/parser/metrics/CommentKeyword.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        "TODO",
+        "Todo",
+        "todo",
+        "HACK",
+        "Hack",
+        "hack",
+        "WTF",
+        "Wtf",
+        "wtf",
```
Comment (ce-bo) [inline src/parser/metrics/CommentKeyword.ts:26]:
```suggestion
        "wtf",
```

File: src/parser/metrics/CommentKeyword.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    ];
+
+    constructor() {
+        for (const keyword of this.keywords) {
+            const regularexpression = "\b" + keyword + "(?![^s.])";
```
Comment (ce-bo) [inline src/parser/metrics/CommentKeyword.ts:31]: Does this really work?

File: src/parser/metrics/CommentKeyword.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        const queryBuilder = new QueryBuilder(language);
+        queryBuilder.setStatements(this.statementsSuperSet);
+
+        const query = queryBuilder.build();
+        let matches: QueryMatch[] = [];
+        if (query !== undefined) {
+            matches = query.matches(tree.rootNode);
+        }
+        for (const keyword of matches) {
+            console.log(keyword.captures.toString());
```
Comment (ce-bo) [inline src/parser/metrics/CommentKeyword.ts:49]: If you want to see the captured text of the match, you can do like this:
```
export function formatCaptures(tree, captures) {
    return captures.map((c) => {
        const node = c.node;
        delete c.node;
        c.text = tree.getText(node);
        return c;
    });
}
```

File: path.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (ce-bo) [inline path.json:1]: This file is not needed, right?

**Review by ce-bo — COMMENTED** [review]
Please add some unit tests for this metric.

File: src/parser/metrics/KeywordsInComments.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        this.#commentLinesCalculator = new CommentLines(allNodeTypes);
+        this.#regexArray = this.#keywords.map((k) => createRegexFor(k));
+    }
+
+    calculate(parsedFile: ParsedFile): MetricResult {
+        const captures = this.#commentLinesCalculator.getQueryCapturesFrom(parsedFile);
+        let metricValue = 0;
+
+        for (const capture of captures) {
+            for (const regex of this.#regexArray) {
```
Comment (clemens-mw) [inline src/parser/metrics/KeywordsInComments.ts:26]: can we use one regex instead (using or e.g.: `/(wtf|todo)/`)?

**Review by clemens-mw — APPROVED** [review]

**Pull request merged**